src/DiscoveryClient.coffee

```diff
@@ -6,40 +6,63 @@ DiscoveryLongPoller = require "./DiscoveryLongPoller"
 ServerList          = require "./ServerList"
 Utils               = require "./Utils"
 Promise             = require "bluebird"
+async               = require "async"
+
 
 class DiscoveryClient
 
-  constructor:(@host, @options)->
+  constructor:(@host, @serviceName, @regionsConfig, @options) ->
+    if not @regionsConfig || not @regionsConfig.discoverRegion || not @regionsConfig.announceRegions
+      throw new Error "Must pass in regionsConfig param with at least one region for discovery apiv2"
```

Comment:
I'm debating just combining serviceName + regionsConfig into just "config" ?  What do you think?